src/daft-plan/src/physical_planner/translate.rs

```diff
@@ -765,156 +768,249 @@ pub fn populate_aggregation_stages(
                 first_stage_aggs
                     .entry(sum_id.clone())
-                    .or_insert(Sum(e.alias(sum_id.clone()).clone()));
+                    .or_insert(AggExpr::Sum(e.alias(sum_id.clone()).clone()));
                 second_stage_aggs
                     .entry(sum_of_sum_id.clone())
-                    .or_insert(Sum(col(sum_id.clone()).alias(sum_of_sum_id.clone())));
+                    .or_insert(AggExpr::Sum(
+                        col(sum_id.clone()).alias(sum_of_sum_id.clone()),
+                    ));
                 final_exprs.push(col(sum_of_sum_id.clone()).alias(output_name));
             }
-            Mean(e) => {
-                let sum_id = Sum(e.clone()).semantic_id(schema).id;
-                let count_id = Count(e.clone(), CountMode::Valid).semantic_id(schema).id;
-                let sum_of_sum_id = Sum(col(sum_id.clone())).semantic_id(schema).id;
-                let sum_of_count_id = Sum(col(count_id.clone())).semantic_id(schema).id;
+            AggExpr::Mean(e) => {
+                let sum_id = AggExpr::Sum(e.clone()).semantic_id(schema).id;
+                let count_id = AggExpr::Count(e.clone(), CountMode::Valid)
+                    .semantic_id(schema)
+                    .id;
+                let sum_of_sum_id = AggExpr::Sum(col(sum_id.clone())).semantic_id(schema).id;
+                let sum_of_count_id = AggExpr::Sum(col(count_id.clone())).semantic_id(schema).id;
                 first_stage_aggs
                     .entry(sum_id.clone())
-                    .or_insert(Sum(e.alias(sum_id.clone()).clone()));
+                    .or_insert(AggExpr::Sum(e.alias(sum_id.clone()).clone()));
                 first_stage_aggs
                     .entry(count_id.clone())
-                    .or_insert(Count(e.alias(count_id.clone()).clone(), CountMode::Valid));
+                    .or_insert(AggExpr::Count(
+                        e.alias(count_id.clone()).clone(),
+                        CountMode::Valid,
+                    ));
                 second_stage_aggs
                     .entry(sum_of_sum_id.clone())
-                    .or_insert(Sum(col(sum_id.clone()).alias(sum_of_sum_id.clone())));
+                    .or_insert(AggExpr::Sum(
+                        col(sum_id.clone()).alias(sum_of_sum_id.clone()),
+                    ));
                 second_stage_aggs
                     .entry(sum_of_count_id.clone())
-                    .or_insert(Sum(col(count_id.clone()).alias(sum_of_count_id.clone())));
+                    .or_insert(AggExpr::Sum(
+                        col(count_id.clone()).alias(sum_of_count_id.clone()),
+                    ));
                 final_exprs.push(
                     (col(sum_of_sum_id.clone()).div(col(sum_of_count_id.clone())))
                         .alias(output_name),
                 );
             }
-            Min(e) => {
+            AggExpr::Stddev(sub_expr) => {
```

Comment:
Maybe leave a comment that we're doing std(X) = sqrt(Var[X]) = sqrt(E[X^2] - E[X]^2)?

Reply:
Left a comment. Will be observable in the next push.